linera-core/src/client/mod.rs

```diff
@@ -3366,6 +3366,55 @@ where
             },
         )
     }
+
+    /// Attempts to update a validator with the local information.
+    #[instrument(level = "trace")]
+    pub async fn sync_validator(
+        &self,
+        remote_node: RemoteNode<P::Node>,
+    ) -> Result<(), ChainClientError> {
+        let validator_chain_state = remote_node
+            .handle_chain_info_query(ChainInfoQuery::new(self.chain_id))
+            .await?;
+        let local_chain_state = self.client.local_node.chain_info(self.chain_id).await?;
+
+        let Some(missing_certificate_count) = local_chain_state
+            .next_block_height
+            .0
+            .checked_sub(validator_chain_state.next_block_height.0)
+        else {
+            debug!("Validator is up-to-date with local state");
+            return Ok(());
+        };
+
+        let missing_certificates_end = usize::try_from(local_chain_state.next_block_height.0)
+            .expect("`usize` should be at least `u64`");
+        let missing_certificates_start = missing_certificates_end
+            - usize::try_from(missing_certificate_count).expect("`usize` should be at least `u64`");
+
+        let missing_certificate_hashes = self
+            .client
+            .local_node
+            .chain_state_view(self.chain_id)
+            .await?
+            .confirmed_log
+            .read(missing_certificates_start..missing_certificates_end)
+            .await?;
+
+        for certificate_hash in missing_certificate_hashes {
+            let certificate = self
+                .client
+                .storage
+                .read_certificate(certificate_hash)
+                .await?;
```

Comment:
The certificates can probably be downloaded in one single operation.
Though I agree that the `handle_confirmed certificate` is likely more time limiting.

Reply:
Oops, I missed that method. Thanks!